\documentclass{article}
\usepackage{amsmath,amssymb}
\usepackage{newpxtext,newpxmath}

\title{The spatial distribution of rare deleterious alleles}
\author{Daniel P. Rice \\
        \and Christian Porras \\
        \and John Novembre}
\date{\today}
\begin{document}

\maketitle

\abstract{[TODO]}

\section*{Introduction}
[TODO]
\section*{Methods}

\subsection*{Overview}
We model the movement, reproduction, and death of the carriers of a rare deleterious allele.
These carriers are generated by mutations in a much larger population of wild-type individuals, whose movements and reproduction we otherwise ignore.
By explicitly modeling only the rare carriers, we can approximate the evolution of their spatial distribution as a \emph{superprocess}.
Recent developments~\cite{recent, developments} in the theory of superprocesses allow us to find the stationary moment generating functional at mutation/selection/migration/drift balance.
% TODO: make sure there aren't older citations for a less general version of the result
Using this result, we can compute the expected (joint) site frequency spectra for arbitrary spatial sampling schemes.

\subsection*{Population genetic model}

We consider a population of organisms living in a habitat $\mathcal{D}$.
We are primarily interested in populations that live in $d-$dimensional continuous habitats ($\mathcal{D} \subset \mathbb{R}^d$) or in networks of discrete demes ($\mathcal{D} \subset \mathbb{Z}$),
but in the following, $\mathcal{D}$ may be an arbitrary Polish space.
Let $\mathcal{N}$ be the population density so that the number of individuals living in a region $A \subset \mathcal{D}$ is given by $N_A = \int_A \mathcal{N}(x) dx$.
% TODO: Clarify that it's a measure?
We assume that $\mathcal{N}$ is large and stably maintained by ecological forces so that we can neglect random density fluctuations due to migration, births, and deaths.
While our model may be extended to include time-varying population densities, here we will focus on spatial variation by computing stationary distributions of time-homogeneous models.

% Maybe move the s stuff later
We are interested in tracking the number and spatial distribution of individuals who carry a rare deleterious variant with fitness cost $s>0$.
For $s \ll 1$, the carriers of the deleterious allele have $1-s$ offspring on average, compared to $1$ for the wild-type. (By focusing on rare alleles, we will neglect dominance effects.)
We assume that wild-type individuals undergo mutation to the deleterious variant at rate $\mu$ per generation.
Thus, we model the influx of de novo mutations as a Poisson point process with intensity $\mu \mathcal{N}(x)$.
% TODO: Sentence still a trouble spot
In general, the density of wild-type individuals is less than the total population density, $\mathcal{N}$, because $\mathcal{N}$ includes carriers.
However, if $\mu \ll s$, the effect of selection dominates the resulting reduction in mutation supply and we can neglect it.

For mathematical tractability, we will assume that carriers of the deleterious allele reproduce, die, and move about the habitat independent of one another and of the background of wild-type individuals.
This assumption is justified as long as the deleterious allele remains rare.
In a continuous habitat---or a lattice with low occupancy---the allele is common in any sufficiently small neighborhood around a carrier.
We can account for this technicality with a suitable definition of `rare'.
To wit, let $\ell$ be the typical distance an individual will move in its lifetime.
We assume that the number of carriers in a ball $B_\ell(x)$ with radius $\ell$ about any point $x \in \mathcal{D}$ is small compared to $N_{B_\ell(x)}$.
% TODO: compare w/ Wright's neighborhood size
As $\ell$ is the shortest length scale in the model, it is reasonable to assume that fluctuations in allele frequency below this scale will be short-lived and not contribute to the long-term evolution of the population.

Given the rare-allele assumption, we model the movement of an individual carrier as a continuous-time Markov process on $\mathcal{D}$ with infinitessimal generator $\mathcal{L}$, independent of the positions of other carriers.
If $\mathcal{D}$ is a set of $n$ discrete demes, $\mathcal{L} = M$, an $n \times n$ matrix of migration rates where $M_{ij}$ is the migration rate from deme $i$ to deme $j$ and $M_{ii}$ is the total rate of migration out of deme $i$.
% TODO: Check that it's not backwards
If $\mathcal{D} \subset \mathcal{R}^d$ and movement is diffusive, $\mathcal{L} = \nabla \cdot D(\mathbf{x}) \nabla$, where $D(\mathbf{x})$ is the local diffusion tensor at position $\mathbf{x} \in \mathcal{D}$.
The dimension of $D$ is $\text{length}^2 / \text{time}$, and if time is measured in generations, $D^{1/2} = \ell$, the typical distance an individual will move in its lifetime.
For translation-invariant, isotropic diffusion, $D$ is a scalar constant and $\mathcal{L} = D\nabla^2$, the Laplacian operator.

Having defined the mutational process by which carriers are generated, and the migration process by which they move around the habitat, it remains to define the mechanism by which they die and reproduce.
Consistent with our assumption that carriers behave independently, we will model reproduction as a continuous-state branching process.
Combining this with our Markov process for movement yields a superprocess model of the evolution of the spatial distribution of carriers.
We now define these processes and then give the main result (Eqs.~\ref{eq:mgf}--\ref{eq:initial_cond}) needed to calculate sample properties.
% TODO - JN: Include citations to other papers that make the same or related approximations here or in intro (E.g. N & Slatkin 2009/10)

\subsubsection*{Continuous-state branching processes}

%TODO: citations

A branching process models a set of particles independently undergoing random ``branching'' events (see~\cite{GrimmettStirzaker:1992}).
When a particle branches, it leaves behind a random number (possibly zero) of identical offspring.
A branching process is thus defined by two parameters: the branching rate, and the distribution of offspring number.
Sample paths of branching processes are functions $N(t):\left[0, \infty\right) \rightarrow \mathbb{N}$ that count the number of extant particles as a function of time.

% JN: This may need to be shortened down the road; itâ€™s instructive but may not fit  in long-run of a concise paper with the results
A continuous-state branching process (CSBP) is a stochastic process that preserves the independent branching property of the discrete branching process, but whose state space is $\mathbb{R}_{\ge 0}$ rather than $\mathbb{N}$.
% TODO: Clarify fractional individuals point
Formally, the CSBP is defined as a limit of branching processes with suitably rescaled time and particle counts so that sample paths converge to right continuous functions with left limits \cite{Lamperti}.
% Citation in LeGall
A CSBP is characterized by a \emph{branching mechanism}:
    $\psi(u) = a u + b u^2$,
where $a$ measures the rate of exponential growth or decay, and $b$ measures the randomness introduced by branching.
In our model, it will be convenient to measure time in generations, which corresponds to setting $b=1$.
In these time units, $a=-s$, the fitness cost of the deleterious mutation.
(More general branching mechanisms are allowed, corresponding to highly skewed offspring number distributions, but we will not consider these here.)

There are two reasons to prefer the CSBP to a discrete branching process.
The first is computational convenience: we can use the known properties of the CSBP (and the corresponding superprocess) to map our problem onto a partial differential equation (Eq.~\ref{eq:mgf}), which we can solve numerically using standard methods.
The second is universality: a CSBP with a particular branching mechanism $\psi$ is the limiting process for a large number of different discrete branching process models.
Rather than specifying the entire distribution of offspring numbers, we need only to specify a single parameter: $s$.
We will see that the unrealistic continuous state is washed out by our finite sampling process: sample properties depend only on the moments of the carrier distribution.
In this sense, the CSBP plays the same role as the Wright-Fisher diffusion does vis-\'a-vis standard population genetic models.
Our approach is thus related to recent methods that project the Wright-Fisher transition density onto a finite basis~\cite{moments, SongSteinrucken:2012}.

\subsubsection*{The superprocess}

If the particles of a branching process move independently through a habitat $\mathcal{D}$ according to a Markov process with generator $\mathcal{L}$, the stochastic process that tracks their number and positions is known as branching diffusion.
Taking the continuous-state limit of a branching diffusion, we arrive at the \emph{superprocess}~\cite{watanabe, dawson}.
For overviews of the superprocess and it properties, see~\cite{dawson, legall, etheridge}.
If we add an influx of particles according to the mutational point process with intensity $\mu \mathcal{N}$, we have a superprocess with immigration~\cite{KawazuWatanabe:1971}.
(Confusingly, in the superprocess literature, our mutation process is called ``immigration'' and the migration process is sometimes called ``mutation''.)

A superprocess $\left\{Z_t\right\}$ is a measure-valued random process.
That is, the value of $Z_t$ at a particular time is a measure on the habitat $\mathcal{D}$.
Measures are defined by how they integrate functions over their domain. Accordingly, we introduce the inner product $\left< Z, f \right> \in \mathbb{R}$, defined as:
\begin{equation}
    \left< Z, f \right> = \int_{\mathcal{D}} f(x) dZ(x),
    \label{eq:inner_product}
\end{equation}
for $Z \in \mathcal{M}_f(\mathcal{D})$, the set of finite measures on $\mathcal{D}$, and $f \in \mathcal{B}_{+}(\mathcal{D})$, the set of nonnegative Borel measurable functions on $\mathcal{D}$~\cite{legall}.
(If $\mathcal{D}$ is discrete, measures are sums of point masses, and the integral is a sum over $\mathbb{Z}$).
In our model, $Z_t$ counts the number of carriers in a region of space.
For a region $A \subseteq \mathcal{D}$, we define the characteristic function $\chi_{A}(x) = 1$ for $x \in A$ and zero otherwise, so that:
\begin{align}
    \left<Z_t, \chi_{A}\right> & = \text{\# of carriers in $A$, and} \\
    \left<Z_t, \chi_{\mathcal{D}}\right> & = \text{total \# of carriers}.
\end{align}
By analogy, we can think of $\left<Z_t, f\right>$ for an arbitrary nonnegative $f$ as counts of carriers weighted by their positions according to $f$.

% JN: Reads like tutorial in places
The probability distribution of a superprocess is characterized by its \emph{moment generating functional} (MGF), $\Phi: \mathcal{B}_{+} \rightarrow \mathbb{R}$, defined as:
\begin{align}
    \Phi_t[f] = \mathbb{E}\left[\exp\left(\left<Z_t, f\right>\right)\right].
    \label{eq:mgf_def}
\end{align}
Just as differentiating the moment generating function of a random variable gives its moments, the functional derivatives of $\Phi_t$ with respect to $f$ give moments of the inner product $\left< Z_t, f\right>$.
Alternatively, $\Phi_t$ is the Laplace transform of the Markov transition kernel of the process.

We are now ready to state the main result that we need from the superprocess literature.
\cite{Stannat} and~\cite{Friesen} have shown that subcritical superprocesses (i.e., ones where $a < 0$ so that the measure decays exponentially) with immigration tend to a stationary distribution, subject to various technical conditions.
In particular, for our process with mutational supply intensity $\mu \mathcal{N}$, Markov generator $\mathcal{L}$, and branching mechanism $\psi(u) = -s(x)u + u^2$,
\begin{align}
    \lim_{t\to \infty}\Phi_t[f] = \Phi[f] \equiv \exp \left( \int_0^\infty \left< \mu \mathcal{N}, u_t \right> dt\right),
    \label{eq:mgf}
\end{align}
where $u_t$ is the solution to the semilinear PDE
\begin{align}
    \frac{\partial}{\partial{t}}u_t(x) & = \mathcal{L}u - su + u^2,
    \label{eq:pde}
\end{align}
subject to initial condition
\begin{align}
    u_0(x) & = f(x).
    \label{eq:initial_cond}
\end{align}
The stationary MGF, $\Phi$, completely characterizes the counts and spatial distribution of carriers of the deleterious alleles of a population at steady-state.
These patterns are due to the balance between the forces of mutation, selection, genetic drift, and migration.
In the next two sections, we will show how Eq.~\ref{eq:mgf} can be used to calculate the expected site frequency spectrum for a spatially localized sample from the population.
The remainder of the Methods are devoted to numerical solution of Eqs.~\ref{eq:pde}--\ref{eq:initial_cond}.

\subsection*{Spatial sampling}

% I don't know if this is the best definition
For a sample of $n$ haploid genomes from the population, not labeled by their geographic origin, define the site frequency spectrum (SFS), $\xi^{(n)}_k$, to be the probability that there are $k$ copies of the deleterious allele.
In a polygenic context, $\xi^{(n)}_k$ corresponds to the expected fraction of sites with $k$ copies of the deleterious allele.
Assuming that the samples are taken independently from the population with replacement, the number of copies of the deleterious allele is the sum of $n$ Bernoulli trials with a random probability of success:
\begin{equation}
    \xi^{(n)}_k = \mathbb{E}\left[ {n \choose k} {P}^k {\left(1-P\right)}^{n-k} \right],
    \label{eq:sfs}
\end{equation}
where $P$ is a random variable representing the probability any particular sampled allele is deleterious.
Thus, the SFS is a linear combination of the moments of $P$, and is completely determined by the moment generating function of $P$.

The distribution of $P$ depends on (1) the locations of carrier individuals characterized by the superprocess $Z$, and (2) the probability we sample a carrier given its location.
For a sample taken uniformly from a region $A \subseteq \mathcal{D}$, $P$ is the fraction of carriers in $A$:
\begin{equation}
    P(A) = \frac{\text{\# of carriers in $A$}}{\text{total \# of individuals in $A$}} = \frac{\left<Z, \chi_A \right>}{N_A}.
\end{equation}

If, instead, the sample is taken by choosing among a countable set of regions $\{ A_i\}$ according to probabilities $\{ \rho_i \}$ and then sampling uniformly within the chosen region,
\begin{align}
    P & = \sum_i \rho_i \frac{\left<Z, \chi_{A_i} \right>}{N_{A_i}}
       = \left< Z, \sum_i \frac{\rho_i \chi_{A_i}}{N_{A_i}} \right>
       \equiv \left< Z, \tilde \rho \right>,
\end{align}
where the second equality uses the linearity of inner products.
For $\mathcal{D} \subset \mathbb{R}^d$, we can take the limit that the sampling probabilities vary continuously, i.e., that $\sum_i \frac{\rho_i \chi_{A_i}}{N_{A_i}} \to \tilde \rho (x) \in \mathcal{B}_{+}(\mathcal{D})$.
If, additionally, the population density is constant, we can write $\tilde \rho(x)$ in terms of a sampling density function as $\rho(x) / \mathcal{N}$.

In any case, the sampling probability is an inner product of the random measure $Z$ with the population-scaled sampling function $\tilde \rho$,
\begin{equation}
    P = \left< Z, \tilde \rho \right>.
    \label{eq:P}
\end{equation}
Therefore, using Eqs.~\ref{eq:mgf_def},~\ref{eq:mgf}, and~\ref{eq:P}, the moment generating function of $P$ at steady-state is given by
\begin{align}
    \phi(z) & = \mathbb{E}\left[ \exp\left(zP\right) \right] \nonumber \\
            & = \mathbb{E}\left[ \exp\left( \left< Z, z \tilde \rho \right> \right)\right] \nonumber \\
            & = \Phi[z \tilde \rho].
            \label{eq:sample_mgf}
\end{align}
% TODO JN: Steps between these lines not totally clear just from Eqn 5
According to Eqs.~\ref{eq:mgf}--\ref{eq:initial_cond}, we can compute the right hand side of Eq.~\ref{eq:sample_mgf} by solving the PDE in Eq.~\ref{eq:pde} with initial condition $u_0 = z \tilde \rho$ and integrating over time.
For small samples, it is practical to use automatic differentiation to compute a truncated power series for $\phi$, and thus the moments of $P$.
The SFS then follows from Eq.~\ref{eq:sfs}.
For large samples, we develop a saddle point approximation to the probability density function of $P$, as described in the next section.
% JN: This part is still not clear.
% Could bring 5-7

\subsubsection*{Geographically labeled samples}

So far, we have considered only samples where no information is retained about the geographic origin of the sampled individuals.
In practice, we often have partial geographic information in the form of, e.g., country or population labels.
For such samples, it is possible to compute a \emph{joint} site frequency spectrum of allele counts in each labeled subsample.
Denoting the vector of sample sizes in each subsample as $\mathbf{n}$ and the deleterious allele counts as $\mathbf{k}$, the joint SFS is given by:
\begin{equation}
    \xi^{\mathbf{n}}_{\mathbf{k}} = \mathbb{E} \left[ \prod_i {n_i \choose k_i} {P_i}^{k_i} {(1-P_i)}^{n_i - k_i} \right],
    \label{eq:jsfs}
\end{equation}
where $\mathbf{P}$ is a random vector of probabilities of sampling the deleterious allele in each subsample, generalizing $P$.

Because the $\{P_i\}$ are all dependent on the underlying random measure $Z$, the expectation does not factorize and we need to find the joint moment generating function,
\begin{align}
    \phi(\mathbf{z}) & = \mathbb{E} \left[ \exp \left(\mathbf{z} \cdot \mathbf{P} \right) \right] \nonumber \\
                     & = \mathbb{E} \left[ \exp \left( \left< Z, \sum_i z_i \tilde{\rho}_i \right> \right) \right] \nonumber \\
                     & = \Phi \left[\sum_i z_i \tilde{\rho}_i \right],
                     \label{eq:jmgf}
\end{align}
where $\{\tilde{\rho}_i\}$ are the population-scaled sampling densities for each sub-sample.
The mixed moments in Eq.~\ref{eq:jsfs} can be found by taking mixed partial derivatives of $\phi$ with respect to $z_i$.

Eqs.~\ref{eq:jsfs}~and~\ref{eq:jmgf} cover the case where the geographic origin of every sampled individual is known.
In that case $n_i = 2$ for all subsamples ($n_i = 1$ for haploids), and $\tilde{\rho}_i$ represents the uncertainty in the sampling location of individual $i$.
Of particular interest is $\mathbf{n} = (2,2)$, which gives the sample configurations for a pair of diploids sampled according to $(\tilde{\rho}_1, \tilde{\rho}_2)$.
Below, we use this case to estimate the genetic variance explained by polygenic scores derived from one samples in region applied to individuals living in another region.

\subsection*{Saddle point approximation to the SFS}
\begin{enumerate}
    \item Problems with moments approach for large samples
    \item SFS approaches the distribution of $P$ anyway
    \item Saddle point approximation in general
    \item Exact application to 1-deme model
    \item $\mu \to 0$ limit
    \item Ansatz
\end{enumerate}

\subsection*{Numerical methods}

\subsubsection*{Solving the cumulant generating functional PDE}
We implement a pseudospectral PDE solver in Julia using the ode library.
We then solve the nondimensionalized PDE with initial data corresponding to the sampling density, integrate the result, and rescale back to the original parameters.

\subsubsection*{Identifying the critical behavior}
We explore the space of initial data to find the boundary of the region where the PDE blows up in finite time.
We then use a logarithmic grid of initial conditions approaching the boundary to extract the critical exponent and parameters.

\subsubsection*{Moment computations}
[TODO] We use julia's autodiff library to compute moments by differentiating the cumulant generating function.
The finite-sample SFS is a polynomial in these moments.

\section*{Results}

\subsection*{Critical behavior of the cumulant generating function}

\subsection*{The spatially sampled SFS}

\subsection*{The decay of explained variance with distance}

\section*{Discussion}

\end{document}
